src/Data/ProgrammesDb/EntityRepository/PromotionRepository.php

```diff
@@ -13,8 +13,12 @@ class PromotionRepository extends EntityRepository
     /**
      * @return array[]
      */
-    public function findActivePromotionsByContext(int $contextId, DateTimeImmutable $datetime, ?int $limit, int $offset): array
+    public function findActivePromotionsByContext(array $ancestryIds, DateTimeImmutable $datetime, ?int $limit, int $offset): array
     {
+        // $contextId is the current context. We want to get both cascading and non-cascading promos for this id
+        // $ancestryIds is an array ids of the parents of the current context. For these ids we only want cascading promos
+        $contextId = array_pop($ancestryIds);
```

Comment:
There is not core_entities with null ancestry values = null. 
Also in the limit case, the coreentity_id (the context_id in here) is the same than the ancestry.

id |   PID        |           Ancestry
11|	b006q2x0|	11,